src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/UnixPkcs12Reader.cs

```diff
@@ -42,10 +42,24 @@ protected void ParsePkcs12(ReadOnlySpan<byte> data)
 
                 unsafe
                 {
-                    IntPtr tmpPtr = Marshal.AllocHGlobal(encodedData.Length);
-                    Span<byte> tmpSpan = new Span<byte>((byte*)tmpPtr, encodedData.Length);
-                    encodedData.CopyTo(tmpSpan);
-                    _tmpManager = new PointerMemoryManager<byte>((void*)tmpPtr, encodedData.Length);
+                    IntPtr tmpPtr = IntPtr.Zero;
+
+                    try
+                    {
+                        tmpPtr = Marshal.AllocHGlobal(encodedData.Length);
```

Comment:
Nit: You can switch this to `NativeMemory.Alloc/Free` while you are on it. It is the prefered unmanaged memory allocator API for modern portable code.